### PR TITLE
[Feature] Support ssl for mysql protocol

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This PR provides the ssl interface framework. Users can implement SSLChannel to support ssl encryption.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlChannel.java [hunk 1]
```diff
@@ -149,14 +196,16 @@ public ByteBuffer fetchOnePacket() throws IOException {
...
+                LOG.info("Receive packet header failed, " +
+                        "remote {} may close the channel.", remoteHostPortString);
                 return null;
             }
             if (packetId() != sequenceId) {
                 LOG.warn("receive packet sequence id[" + packetId() + "] want to get[" + sequenceId + "]");
                 throw new IOException("Bad packet sequence.");
             }
             int packetLen = packetLen();
+            LOG.info("packetLen is {}", packetLen);
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/mysql/MysqlChannel.java:208]: remove debug message

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlProto.java [hunk 2]
```diff
@@ -98,21 +98,38 @@ public static boolean negotiate(ConnectContext context) throws IOException {
...
+                LOG.info("start to enable ssl connection");
+                if (!context.enableSSL()) {
+                    LOG.warn("enable ssl connection failed");
+                    ErrorReport.report(ErrorCode.ERR_CHANGE_TO_SSL_CONNECTION_FAILED);
+                    sendResponsePacket(context);
+                    return false;
+                } else {
+                    LOG.info("enable ssl connection successfully");
+                }
+            } catch (IOException e) {
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/mysql/MysqlProto.java]: done

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlChannel.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/mysql/MysqlChannel.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 124 (43.55%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/mysql/ssl/SSLChannelImpClassLoader.java|0|17|00.00%|[14, 15, 17, 22, 23, 26, 27, 28, 29, 32, 33, 34, 35, 36, 37, 38, 39]|
|:large_blue_circle:|com/starrocks/qe/QeService.java|4|20|20.00%|[49, 50, 55, 76, 77, 78, 80, 81, 83, 84, 85, 87, 88, 90, 91, 92]|
|:large_blue_circle:|com/starrocks/mysql/MysqlChannel.java|12|31|38.71%|[131, 134, 139, 140, 144, 145, 147, 148, 149, 150, 153, 154, 155, 156, 157, 158, 163, 199, 242]|
|:large_blue_circle:|com/starrocks/mysql/MysqlProto.java|10|24|41.67%|[113, 114, 115, 116, 117, 118, 120, 124, 125, 126, 212, 216, 217, 218]|
|:large_blue_circle:|com/starrocks/mysql/MysqlHandshakePacket.java|3|5|60.00%|[76, 77]|
|:large_blue_circle:|com/starrocks/mysql/MysqlAuthPacket.java|2|3|66.67%|[89]|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|6|7|85.71%|[599]|
|:large_blue_circle:|com/starrocks/mysql/nio/AcceptListener.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/nio/NMysqlServer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/MysqlCapability.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/nio/NConnectContext.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/nio/NMysqlChannel.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/MysqlServer.java|3|3|100.00%|[]|

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Review by padmejin — DISMISSED** [review]
LGTM

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 52 / 117 (44.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/mysql/ssl/SSLChannelImpClassLoader.java|0|17|00.00%|[14, 15, 17, 22, 23, 26, 27, 28, 29, 32, 33, 34, 35, 36, 37, 38, 39]|
|:large_blue_circle:|com/starrocks/qe/QeService.java|4|15|26.67%|[48, 49, 54, 75, 76, 77, 79, 80, 82, 83, 84]|
|:large_blue_circle:|com/starrocks/mysql/MysqlChannel.java|12|31|38.71%|[131, 134, 139, 140, 144, 145, 147, 148, 149, 150, 153, 154, 155, 156, 157, 158, 163, 199, 242]|
|:large_blue_circle:|com/starrocks/mysql/MysqlProto.java|10|24|41.67%|[113, 114, 115, 116, 117, 118, 120, 124, 125, 126, 212, 216, 217, 218]|
|:large_blue_circle:|com/starrocks/mysql/MysqlHandshakePacket.java|3|5|60.00%|[76, 77]|
|:large_blue_circle:|com/starrocks/mysql/MysqlAuthPacket.java|2|3|66.67%|[89]|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|6|7|85.71%|[599]|
|:large_blue_circle:|com/starrocks/mysql/nio/AcceptListener.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/nio/NMysqlServer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/MysqlCapability.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/nio/NConnectContext.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/nio/NMysqlChannel.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/MysqlServer.java|3|3|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Comment by gengjun-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -1755,4 +1755,13 @@ public class Config extends ConfigBase {
...
     public static boolean recursive_dir_search_enabled = false;
+
+    @ConfField
+    public static String ssl_keystore_location = "";
+
+    @ConfField
+    public static String ssl_keystore_password = "";
+
+    @ConfField
+    public static String ssl_key_password = "";
```
Comment (decster) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1766]: Will this information be exposed in web UI or other metrics interface?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1766]: This is an enterprise-edition  feature. So these configs will not exposed in our community docs.

Comment (decster) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1766]: I mean `http://<FE_HOST>:8030/variable`

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 4]
```diff
@@ -1755,4 +1755,13 @@ public class Config extends ConfigBase {
 
     @ConfField(mutable = true)
     public static boolean recursive_dir_search_enabled = false;
+
+    @ConfField
+    public static String ssl_keystore_location = "";
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1760]: Add more clear commit message.

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 5]
```diff
@@ -1755,4 +1755,13 @@ public class Config extends ConfigBase {
...
+
+    @ConfField
+    public static String ssl_keystore_password = "";
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1763]: If a user wants to enable SSL, it needs to restart FE?

File: fe/fe-core/src/main/java/com/starrocks/mysql/ssl/SSLChannelImpClassLoader.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+public class SSLChannelImpClassLoader {
+    private static final Logger LOG = LogManager.getLogger(Auth.class);
+    public static final String SSL_CHANNEL_CLASS_NAME = "com.starrocks.plugins.ssl.SSLChannelImp";
+    public static final String SSL_CHANNEL_JAR_PATH = StarRocksFE.STARROCKS_HOME_DIR + "/lib/starrocks-ssl-1.0.jar";
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/mysql/ssl/SSLChannelImpClassLoader.java:17]: Where to place the jar?

**Pull request closed without merging**